### PR TITLE
Fix copying ETCd backups between OCS buckets

### PR DESCRIPTION
**Pull request by Wieneo**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
The proposed changes (in combination with https://github.com/gardener/etcd-backup-restore/pull/465) enable etcd-druid to successfully copy ETCd backups between OCS buckets.
It utilizies the new way of supplying access information to the etcd-backup-restore job (mounting a secret to a specific directory).

**Release note**:

```other operator
ETCd backups can now be successfully copied between OCS buckets.
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @Wieneo for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/images.yaml
Comment (kris94) [inline charts/images.yaml:5]: I guess this should be the release that contains the [this PR](https://github.com/gardener/etcd-backup-restore/pull/465/).  I think it would be 1.16.something, so lets leave this as TODO for now.

/hold

**Review by kris94 — CHANGES_REQUESTED** [review]
Thank you for your contribution!
Please rebase this PR to the latest master and drop all commits that are not part of this change.

**Comment by Wieneo** [comment]
@kris94 I'm sorry, I messed up the PR somewhere.
I've rebased it and dropped all erroneous commits.

**Comment by ishan16696** [comment]
Can you also update the [etcd-statefulset.yaml for OCS](https://github.com/gardener/etcd-druid/blob/master/charts/etcd/templates/etcd-statefulset.yaml#L297-L329) ?

**Comment by Wieneo** [comment]
@ishan16696 I've updated the manifest :)

**Comment by ishan16696** [comment]
Can you also update this file as well https://github.com/gardener/etcd-druid/blob/master/controllers/compaction_lease_controller.go#L450-L455 , you can take a look at [this](https://github.com/gardener/etcd-druid/pull/301/files#diff-f45c309f86c388357af7d27e5cd09bc1fd1455535ed1ffeeb4f39b61cc9bc928) for reference

**Review by ishan16696 — APPROVED** [review]
LGTM!! . We will merge this PR once backup-restore `v0.17.0` get release which contains this PR https://github.com/gardener/etcd-backup-restore/pull/465

File: charts/etcd-copy-backups/templates/etcd-copy-backups-job.yaml
Comment (timuthy) [inline charts/etcd-copy-backups/templates/etcd-copy-backups-job.yaml:90]:
```suggestion
{{- else if eq .Values.targetStore.storageProvider "OCS" }}
```

**Review by timuthy — CHANGES_REQUESTED** [review]
Thanks a lot for the PR. Can you please take a look at the comment I left in-line?

**Comment by Wieneo** [comment]
I've commited your changes :)

**Review by abdasgupta — APPROVED** [review]

**Pull request merged**
